### PR TITLE
[RLlib; CI] Further split RLlib CI jobs.

### PR DESCRIPTION
**Pull request by sven1977**
Further split RLlib's CI jobs to bring down total testing time per-PR to ~20min.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm.py [hunk 1]
```diff
@@ -1994,13 +1994,13 @@ def merge_trainer_configs(
         """
         config1 = copy.deepcopy(config1)
         if "callbacks" in config2 and type(config2["callbacks"]) is dict:
-            legacy_callbacks_dict = config2["callbacks"]
-
-            def make_callbacks():
-                # Deprecation warning will be logged by DefaultCallbacks.
-                return DefaultCallbacks(legacy_callbacks_dict=legacy_callbacks_dict)
+            deprecation_warning(
```
Comment (sven1977) [inline rllib/algorithms/algorithm.py:1997]: Sorry, these shouldn't be here. Let me fix!

Comment (sven1977) [reply rllib/algorithms/algorithm.py]: done

File: .buildkite/pipeline.ml.yml [hunk 2]
```diff
@@ -32,7 +44,7 @@
       --test_arg=--framework=tf
       rllib/...
 
-- label: ":brain: RLlib: Learning discr. actions TF2-eager-tracing"
+- label: ":brain: RLlib: Learning discr. actions (I) TF2-eager-tracing"
```
Comment (ArturNiederfahrenhorst) [inline .buildkite/pipeline.ml.yml:47]: Don't we have to exclude the cartpole tests here?

File: .buildkite/pipeline.ml.yml [hunk 3]
```diff
@@ -56,7 +80,7 @@
       --test_arg=--framework=tf2
       rllib/...
 
-- label: ":brain: RLlib: Learning discr. actions PyTorch"
+- label: ":brain: RLlib: Learning discr. actions (I) PyTorch"
```
Comment (ArturNiederfahrenhorst) [inline .buildkite/pipeline.ml.yml:83]: Don't we have to exclude the cartpole tests here?

**Review by ArturNiederfahrenhorst — COMMENTED** [review]
Awesome! I think you might have forgotten exclusion tags in the torch and eager test though!

File: .buildkite/pipeline.ml.yml (hunk 3 above)
Comment (sven1977) [reply .buildkite/pipeline.ml.yml:83]: Good catch! Fixed.

File: .buildkite/pipeline.ml.yml (hunk 2 above)
Comment (sven1977) [reply .buildkite/pipeline.ml.yml:47]: Good catch! Fixed.

**Review by ArturNiederfahrenhorst — APPROVED** [review]
Approved on condition of green tests

**Pull request merged**